pkg/segment/segment.go

```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	}
 
 	// puts pointer to pointerDB
-	_, err = s.pc.Put(ctx, &pr)
+	_, err = s.pdb.Put(ctx, &pr)
 	if err != nil || status.Code(err) == codes.Internal {
 		zap.L().Error("failed to put", zap.Error(err))
 		return Error.Wrap(err)
 	}
 	return nil
 }
 
-// Get retrieves a file from the erasure code client with help from overlay and pointerdb
+// Get retrieves a file using erasure code, overlay, and pointerdb clients
 func (s *segmentStore) Get(ctx context.Context, path paths.Path) (ranger.Ranger, Meta, error) {
-	m := Meta{
-		Data: []byte(""),
+	m := Meta{}
+	// TODO: remove this chunk after pointerdb client interface merged
+	gr := &ppb.GetRequest{
+		Path:   []byte(fmt.Sprintf("%s", path)),
+		APIKey: nil,
+	}
+
+	pdbRes, err := s.pdb.Get(ctx, gr)
+	if err != nil {
+		return nil, m, err
+	}
+
+	// TODO: remove this chunk after pointerdb client interface merged
+	pointer := &ppb.Pointer{}
+	err = proto.Unmarshal(pdbRes.Pointer, pointer)
+	if err != nil {
+		return nil, m, err
+	}
+
+	if pointer.Type != ppb.Pointer_REMOTE {
+		zap.L().Error("TODO: only getting remote pointers supported")
+		return nil, m, err
+	}
+
+	remoteSeg := pointer.Remote
+	var nodes []*opb.Node
+	for i := 0; i < len(remoteSeg.RemotePieces); i++ {
+		overlayRes, err := s.oc.Lookup(ctx, &opb.LookupRequest{NodeID: remoteSeg.RemotePieces[i].NodeId})
```

Comment:
shoot, would be nice to do this in parallel

even better would be if the pointerdb response had gotten this information from overlaycache for us.

no need to fix for now